mi/dataset/parser/metbk_a_dcl.py

```diff
@@ -140,68 +153,34 @@ def parse_file(self):
-        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(shortwave_irradiance, 'shortwave_irradiance', SENSOR_GROUP_SHORTWAVE_IRRADIANCE))
-        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(eastward_wind_velocity, 'eastward_wind_velocity', SENSOR_GROUP_EASTWARD_WIND_VELOCITY))
-        INSTRUMENT_PARTICLE_MAP.append(self.add_particle(northward_wind_velocity, 'northward_wind_velocity', SENSOR_GROUP_NORTHWARD_WIND_VELOCITY))
-
-    @staticmethod
-    def add_particle(raw_data, particle, group):
-        """
-        This function will add particle to the instrument particle map
-        """
-        if isinstance(raw_data, float):
-            return particle, group, float
-        else:
-            return particle, group, str
+                if not re.findall(r'\[.*\]:[a-z|A-Z]+', line):                  # Disregard anything that has a character after [Something]:
+                    line = re.sub(r'\[.*\]:', '', line)
+                    raw_data = line.split()
+                    raw_data[0:2] = [' '.join(raw_data[0:2])]                   # Merge the first and second elements to form a timestamp
+                    if raw_data is not None:
+                        for i in range(1, len(raw_data)):                       # Ignore 0th element, because that is the timestamp
+                            raw_data[i] = self.select_type(raw_data[i])
+                    # self.construct_instrument_particle_map(raw_data)
+                    particle = self._extract_sample(particle_class,
+                                                    None,
+                                                    raw_data,
+                                                    preferred_ts=DataParticleKey.PORT_TIMESTAMP)
+                    self._record_buffer.append(particle)
+
+                # else:
+                #     # If it's a valid metadata record, ignore it.
+                #     # Otherwise generate warning for unknown data.
+                #     error_message = 'Unknown data found in chunk %s' % line
+                #     log.warn(error_message)
+                #     self._exception_callback(UnexpectedDataException(error_message))
 
     @staticmethod
     def select_type(raw_list_element):
         """
         This function will return the float or string value of the particle.
         """
-        if raw_list_element is not isinstance(raw_list_element, float):
-            if re.match(r'\[.*\]:', raw_list_element) is not None:          # Get rid of the [metbk:DLOGP6]: (or similar) in front of float number
-                return float(re.sub(r'\[.*\]:', '', raw_list_element))
-            try:
-                return float(raw_list_element)
-            except ValueError:
-                return str(raw_list_element)
+        try:
+            return float(raw_list_element)
+        except ValueError:
+            pass
```

Comment:
what gets returned in the case of a ValueError? You're setting a value equal to the return, so probably should return `None`

Reply:
This returns a None, but I think that is a good idea that I should explicitly return `None`